core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/OAbstractPaginatedStorage.java

```diff
@@ -2011,6 +2023,13 @@ public void preallocateRids(final OTransactionInternal clientTx) {
           database.getMetadata().clearThreadLocalSchemaSnapshot();
         }
       } finally {
+        if (pessimisticLock) {
+          for (ORecordOperation recordOperation : recordOperations) {
+            if (recordOperation.type == ORecordOperation.UPDATED || recordOperation.type == ORecordOperation.DELETED) {
+              releaseWriteLock(recordOperation.getRID());
+            }
+          }
+        }
         stateLock.releaseReadLock();
```

Comment:
I think this like should be in the `finally{}` block in case the release above fails.

Reply:
you mean wrap the record lock release in a try{} finally {} ? in terms of business the release here should never fail, it may do due to a bug